packages/api/cms-api/src/auth/resolver/auth.resolver.ts

```diff
@@ -48,6 +54,27 @@ export function createAuthResolver(config?: AuthResolverConfig): Type<unknown> {
         permissionsForScope(@Parent() user: CurrentUser, @Args("scope", { type: () => GraphQLJSONObject }) scope: ContentScope): string[] {
             return user.permissions.map((p) => p.permission).filter((permission) => this.accessControlService.isAllowed(user, permission, scope));
         }
+
+        @ResolveField(() => [GraphQLJSONObject])
+        allowedContentScopes(@Parent() user: CurrentUser): ContentScope[] {
+            return uniqWith(
+                user.permissions.flatMap((p) => p.contentScopes),
+                isEqual,
+            );
+        }
+
+        @ResolveField(() => [GraphQLJSONObject])
+        async allowedContentScopesWithLabels(@Parent() user: CurrentUser): Promise<ContentScopeWithLabel[]> {
+            const allowedContentScopes = this.allowedContentScopes(user);
+            return (await this.service.getAvailableContentScopesWithLabels()).filter((contentScopeWithLabels) =>
+                allowedContentScopes.some((allowedContentScope) =>
+                    isEqual(
+                        Object.fromEntries(Object.entries<{ value: string }>(contentScopeWithLabels).map(([key, value]) => [key, value.value])),
+                        allowedContentScope,
+                    ),
+                ),
+            );
```

Comment:
This is hard to read. Could it be made easier by iterating allowedContentScopes and finding the label from `this.service.getAvailableContentScopesWithLabels()`?